src/memory/process.rs

```diff
@@ -106,6 +107,104 @@ pub trait ProcessTraits where Self: Sized {
         buff: &mut [u8]
     ) -> Result<(), ProcessError>;
 
+    fn read_struct<T: Sized>(
+        &self,
+        addr: usize
+    ) -> Result<T, ProcessError> {
+        let mut buff = vec![0u8; size_of::<T>()];
+
+        let byte_buff = unsafe {
+            std::slice::from_raw_parts_mut(
+                buff.as_mut_ptr() as *mut u8,
+                buff.len()
+            )
+        };
+
+        self.read(addr, byte_buff.len(), byte_buff)?;
+
+        let s: T = unsafe { std::ptr::read(buff.as_ptr() as *const _) };
+
+        Ok(s)
+    }
+
+    fn read_struct_array<T: Sized>(
+        &self,
+        addr: usize,
+        len: usize
+    ) -> Result<Vec<T>, ProcessError> {
+        let size = size_of::<T>() + align_of::<T>();
+        let mut buff = vec![0u8; size * len];
+
+        let mut byte_buff = unsafe {
+            std::slice::from_raw_parts_mut(
+                buff.as_mut_ptr() as *mut u8,
+                buff.len()
+            )
+        };
+        
+        self.read(addr, byte_buff.len(), byte_buff)?;
+
+        let mut arr = Vec::with_capacity(len);
+
+        while byte_buff.len() >= size_of::<T>() {
+            let (head, tail) = byte_buff.split_at_mut(size);
+            let s: T = unsafe { std::ptr::read(head.as_ptr() as *const T) };
+            arr.push(s);
+            byte_buff = tail;
+        }
+
+        Ok(arr)
```

Comment:
Same alignment and heap allocation issue here. Could be replaced with this:
```rs
let mut buff: Vec<_> = std::iter::repeat_with(std::mem::MaybeUninit::<T>::uninit)
    .take(len)
    .collect();

let ptr: *mut u8 = buff.as_mut_ptr().cast();
let slice = unsafe { std::slice::from_raw_parts_mut(ptr, size_of::<T>() * len) };

self.read(addr, slice.len(), slice)?;

let ptr: *mut T = buff.as_mut_ptr().cast();
let len = buff.len();
let cap = buff.capacity();

std::mem::forget(buff);

Ok(unsafe { Vec::from_raw_parts(ptr, len, cap) })
```
Again, miri could help ensuring there's nothing wrong with the unsafety and marking the whole method as unsafe might be in order.